### PR TITLE
SJSU-1054_1: City Overview (Frontend)

### PR DESCRIPTION
**Pull request by DaleCS**
**Story description:**
A city overview including city pictures, description of city, attractions in city, weather chart per month.

**What I did:**
_ Designed page for city overview.
_ Wired up action-redux workflow to React component.
_ Connected OpenWeatherMap API to redux.

**How to test:**
_ Go to localhost:3000/cities/0
_ Run the app. See if there are any errors caused by my code.
_ Look at the page using varying browser sizes to test responsiveness.
_ Look at my code. Let me know if there's ambiguity or incorrect usage of react, redux, etc..
_ Check if the story requirements are met.

### CODE REVIEW

File: front_end/src/actions/cityActions.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+} from "./types";
+
+// API key imports
+import { weatherMapAPIKey } from "../apiKeys/keys";
+
+// Fetches city data using the provided city id
+export let fetchCityById = cityId => dispatch => {
+  setLoadingCity();
+
+  // TODO: Uncomment and fill in missing code below once backend API is implemented then remove prototype code
```
Comment (SandroSal) [inline front_end/src/actions/cityActions.js]: how far along is the back? would be good to merge the actual code... not a prototype and then needing to go back and change again

**Comment by SandroSal** [comment]
/cities/0 for testing?

when I open that, the Abode logo in the header disappears? lol

also, it'd be nice to have a chart of the temperature per month like
Jan 65
Feb 65
Mar 70
Apr 74
May 77
Jun 80
Jul 90
....

**Comment by DaleCS** [comment]
> 
> 
>     * abode logo disappears when opening cities page
> 
>     * bottom looks a little crammed, maybe add some more spacing before and after the weather boxes
> 
>     * doesn't show today's weather...
> 
> 
> can you please add redux to connect landing and cities pages?

Requested changes applied.

**Comment by DaleCS** [comment]
>     * weather container now shows infinite spinner
> 
>     * please also link the inspiration pic to cities ("Explore XYZ")
> 
>     * merge conflict in types.js
> 
> 
> Lance said he would finish the backend tonight or tomorrow the latest. Let's wait for his PR so we can finish this story at once and don't have to start over...

_ Might possibly be an internet issue. 
_ Done
_ I don't see any merge conflicts with dev-general...

File: front_end/src/actions/cityActions.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  /*
+  axios.get(BACKEND API CODE)
```
Comment (HongbinZheng) [inline front_end/src/actions/cityActions.js]: route would be `/api/cityView/city/cityId`

**Pull request merged**